node/services/bn-manager.go

```diff
@@ -73,10 +79,49 @@ func (m *BeaconClientManager) GetClientTypeName() string {
 
 func (m *BeaconClientManager) SetPrimaryReady(ready bool) {
 	m.primaryReady = ready
+	if ready {
+		m.primaryFailTime = time.Time{}
```

Comment:
instead of `primaryFailTime` and `fallbackFailTime` as fields on `BeaconClientManager` you could use a unexported struct which has a boolean field to indicate failure (rather than relying on 0 value to mean success)